File: Source/Library/PrxSample/prx/src/main/java/com/philips/cdp/prxclient/request/ProductAssetRequest.java
package com.philips.cdp.prxclient.request;

import com.philips.cdp.prxclient.datamodels.assets.AssetModel;
import com.philips.cdp.prxclient.response.ResponseData;

import org.json.JSONObject;

import java.util.Map;

/**
 * Description :
 * Project : PRX Common Component.
 * Created by naveen@philips.com on 02-Nov-15.
 */
public class ProductAssetRequest extends PrxRequest {

    private static final String PRX_REQUEST_URL = "https://%s/product/%s/%s/%s/products/%s.assets";
    private String mCtn = null;
    private String mRequestTag = null;

    public ProductAssetRequest(String ctn, String requestTag) {
        this.mCtn = ctn;
        this.mRequestTag = requestTag;
    }

    @Override
    public ResponseData getResponseData(JSONObject jsonObject) {
        return new AssetModel().parseJsonResponseData(jsonObject);
    }

    @Override
    public String getRequestUrl() {
        return String.format(PRX_REQUEST_URL, getServerInfo(), getSector(), getLocaleMatchResult(),
                getCatalog(), mCtn);
    }

    @Override
    public int getRequestType() {
        return RequestType.GET.getValue();
    }

    @Override
    public Map<String, String> getHeaders() {
        return null;
    }

    @Override
    public Map<String, String> getParams() {
        return null;
    }
}

File: Source/prx/Source/Library/PrxSample/prx/src/main/java/com/philips/cdp/prxclient/request/ProductAssetRequest.kt
package com.philips.cdp.prxclient.request

import com.philips.cdp.prxclient.PrxConstants
import com.philips.cdp.prxclient.datamodels.assets.AssetModel
import com.philips.cdp.prxclient.response.ResponseData
import org.json.JSONObject

/**
 * The type Product asset request.
 */
class ProductAssetRequest : PrxRequest {
    private var mRequestTag: String? = null

    /**
     * Instantiates a new Product asset request.
     * @since  1.0.0
     * @param ctn        product ctn
     * @param requestTag requestTag
     */
    constructor(ctn: String?, requestTag: String?) : super(ctn, PRXAssetAssetServiceID) {
        mRequestTag = requestTag
    }

    /**
     * Instantiates a new Product asset request.
     * @since 1.0.0
     * @param ctn         product ctn
     * @param sector      sector
     * @param catalog     catalog
     * @param requestTag  request tag
     */
    constructor(ctn: String?, sector: PrxConstants.Sector?, catalog: PrxConstants.Catalog?, requestTag: String?) : super(ctn, PRXAssetAssetServiceID, sector, catalog) {
        mRequestTag = requestTag
    }

    override fun getResponseData(jsonObject: JSONObject?): ResponseData? {
        return AssetModel().parseJsonResponseData(jsonObject)
    }

    companion object {
        private const val PRXAssetAssetServiceID = "prxclient.assets"
    }
}